cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/queue/JobQueueDataset.java

```diff
@@ -370,6 +370,8 @@ public void persistSubscriberState(String topic, String messageId) {
   public static void setupDatasets(DatasetFramework datasetFramework) throws IOException,
     DatasetManagementException {
     datasetFramework.addInstance(JobQueueDataset.class.getName(), Schedulers.JOB_QUEUE_DATASET_ID,
-                                 DatasetProperties.EMPTY);
+                                 DatasetProperties.builder().
+                                   add(Constants.Scheduler.JOB_QUEUE_NUM_PARTITIONS, DEFAULT_NUM_PARTITIONS)
```

Comment:
why is this needed if it is the default?

Reply:
Its because https://github.com/caskdata/cdap/blob/release/5.1/cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/queue/JobQueueDatasetDefinition.java#L48-L50 - expects it to be present